code/game/machinery/recharger.dm

```diff
@@ -269,6 +269,20 @@
 /obj/machinery/recharger/self_powered	//ideal for the Thunderdome
 	self_powered = 1
 
+/obj/machinery/recharger/kick_act(mob/living/carbon/human/H)
+	. = ..()
+	if(charging)
+		if(prob(25))
+			visible_message("<span class='notice'>\The [charging] pops out of \the [src]!</span>")
```

Comment:
instead of pasting this, we should probably just have a proc for dropping the contained gun